models/ed/R/write.configs.ed.R

```diff
@@ -377,30 +383,44 @@ write.config.xml.ED2 <- function(settings, trait.values, defaults = settings$con
       # happen is for there to be two settings for each PFT: the 'num' to use to represent the PFT to
       # ED, and the 'defaults.PFT' (name or number) to use for pulling default parameter values.
       pft.number <- pftmapping$ED[which(pftmapping == pft)]
-      if (length(pft.number) == 0) {
+      
+      if(pft=="soil"){
+        data(soil)
```

Comment:
If this is coming from inside the package, I would specify it explicitly. `data(soil, package = "packagename")`.